node/declaration.ts

```diff
@@ -116,20 +116,73 @@ const createDeclarationsFromFunctionConfigs = (
   return declarations
 }
 
-// Validates and normalizes a pattern so that it's a valid regular expression
-// in Go, which is the engine used by our edge nodes.
-export const parsePattern = (pattern: string) => {
+/**
+ * Validates and normalizes a pattern so that it's a valid regular expression
+ * in Go, which is the engine used by our edge nodes.
+ *
+ * @param pattern Original regular expression
+ * @param extractedExclusionPatterns If set, negative lookaheads (which are
+ * not supported in Go) are transformed into a list of exclusion patterns to
+ * be stored in this array
+ * @returns Normalized regular expression
+ */
+export const parsePattern = (pattern: string, extractedExclusionPatterns?: string[]) => {
```

Comment:
nit: Could we return `extractedExclusionPatterns` as an array, instead of mutating the parameter?

Reply:
That would be the cleanest way, for sure. But `parsePattern` is used in a few different contexts and I wanted to change the API surface as little as possible to contain the change behind the flag. We can revisit this in the future.